content/websocket-sample.md

```diff
@@ -0,0 +1,190 @@
+
+## Prerequisites
+
+You will need a Kubernetes cluster with Fission installed.
+Please follow the [instructions here][5] for installing Fission in your Kubernetes cluster.
+Then, verify that the fission cli is working by using the `fission --version` command.
+
+## Try it out
+
+In order to try it out, you can either use fission specs provided with the example or manually generate the specs.
+We will cover both the approaches in this blog.
+
+### Use existing specs
+
+- Clone the [fission/examples][6] repository and navigate to websocket sample code.
+
+```
+git clone https://github.com/fission/examples.git
+cd examples/samples/websocket
+```
+
+- There you must have the `specs` directory. 
+We are going to use these specs and deploy our function.
+
+```
+fission spec apply
+
+DeployUID: 1df9464b-5f73-4623-8187-a2f431d5c828
+Resources:
+ * 1 Functions
+ * 1 Environments
+ * 1 Packages
+ * 1 Http Triggers
+ * 0 MessageQueue Triggers
+ * 0 Time Triggers
+ * 0 Kube Watchers
+ * 1 ArchiveUploadSpec
+Validation Successful
+1 environment created: nodejs
+1 package created: broadcast-pkg
+1 function created: broadcast
+1 HTTPTrigger created: broadcast
+```
+
+Now that we have deployed the function, let's [test our chat application](#chat-application).
+
+### Manually generate specs
+
+- Create a directory as `websocket`.
+
+```
+mkdir websocket && cd websocket
+```
+
+- Get websocket sample code (`broadcast.js` and `index.html`).
+
+```
+wget https://raw.githubusercontent.com/fission/examples/master/samples/websocket/broadcast.js
+wget https://raw.githubusercontent.com/fission/examples/master/samples/websocket/index.html
+```
+
+- Create an initial declarative application specification.
+The command will generate a `specs` directory and a fission deployment config spec file.
+
+```
+fission spec init
+```
+
+- Create fission [environment][7] for the function.
+
+```
+fission env create \
+ --name=nodejs \
+ --image=fission/node-env:latest \
+ --spec
+```
+
+- Create spec for the broadcast [function][8].
+
+```
+fission fn create \
+ --name=broadcast \
+ --env=nodejs \
+ --rpp=5 \
+ --deploy=broadcast.js \
+ --spec
+```
+
+- Create spec for [HTTP trigger][4].
+
+```
+fission httptrigger create \
+ --name=broadcast \
+ --url=/broadcast \
+ --function=broadcast \
+ --spec
+```
+
+- Now, we can validate and apply the generated declarative application specifications.
+
+```
+fission spec validate
+fission spec apply
+```
+
+### Chat application
+
+- The fission router is exposed as a service.
+If you are running a kind cluster, you must get the node internal IP and router service port.
+
+```
+kubectl get nodes -o wide
+kubectl get svc -n fission -l application=fission-router
+```
+
+- Edit the `index.html` and update the connection URL at line `32`.
+
+```
+...
+conn = new WebSocket("ws://<node-internal-ip>:<router-svc-node-port>/broadcast");
+...
+```
+
+- Run the HTTP server that will serve the chat application.
+
+```
+python3 -m http.server 8080
+```
```

Comment:
It be ideal is if the user does not have to run a server manually locally. You can write a simple UI function which is deployed inside Fission itself and that one connects to Backend NodeJS App. The bank sample repo has examples of how to deliver a HTML file using a fission function!